### PR TITLE
include sn-react-typescript-boilerplate #402

### PR DESCRIPTION
**Pull request by sumirart**
I want to resolve issue #402

TODO:
 * [x]  Create a folder named sn-react-typescript-boilerplate under examples folder
 * [x]  Copy the relevant files to there (src,test, .gitignore, LICENSE, README.md, index.html, package.json, tsconfig.json, webpack.config.js)
 * [x]  Remove devDependencies that are not needed (eslint related packages, enzyme related packages, husky, lint staged)
 * [x]  Upgrade to latest version of sensenet scoped packages
 * [x]  Remove fields that are not needed from package.json (husky, lint-staged, jest)
 * [x]  Edit tsconfig.json to extend from the tsconfig-base.json from the sn-client root
 * [x]  Try to run it & build it
 * [x]  Edit articles that references [sn-react-typescript-boilerplate](https://github.com/SenseNet/sn-react-typescript-boilerplate) in [sensenet.github.io](https://github.com/SenseNet/sensenet.github.io)

(I'm new to this so any advice is very welcome, thanks!)

### CODE REVIEW

**Comment by B3zo0** [comment]
Awsome work @sumirart thanks.

Can you please add .DS_Store to the .gitignore files and remove the added .DS_Store?

File: examples/sn-react-typescript-boilerplate/package-lock.json [hunk 1]
```diff
@@ -0,0 +1,10491 @@
+{
```
Comment (B3zo0) [inline examples/sn-react-typescript-boilerplate/package-lock.json]: This file is not needed. Please remove it.

File: examples/sn-react-typescript-boilerplate/package.json [hunk 2]
```diff
@@ -0,0 +1,66 @@
...
+  },
+  "private": true,
+  "publishConfig": {
+    "access": "private"
+  },
+  "author": "Sense/Net <snteam@sensenet.com> (http://sensenet.com/)",
+  "license": "GPL-2.0",
+  "homepage": "https://sensenet.com",
+  "devDependencies": {
+    "@types/enzyme": "^3.9.3",
```
Comment (B3zo0) [inline examples/sn-react-typescript-boilerplate/package.json]:
```suggestion
```

File: examples/sn-react-typescript-boilerplate/package.json [hunk 3]
```diff
@@ -0,0 +1,66 @@
...
+    "@types/jest": "^24.0.13",
```
Comment (B3zo0) [inline examples/sn-react-typescript-boilerplate/package.json]:
```suggestion
```

File: examples/sn-react-typescript-boilerplate/package.json [hunk 4]
```diff
@@ -0,0 +1,66 @@
...
+    "@types/react": "^16.8.19",
+    "@types/react-dom": "^16.8.4",
+    "@typescript-eslint/eslint-plugin": "^2.0.0",
```
Comment (B3zo0) [inline examples/sn-react-typescript-boilerplate/package.json]:
```suggestion
```

File: examples/sn-react-typescript-boilerplate/package.json [hunk 5]
```diff
@@ -0,0 +1,66 @@
...
+    "@typescript-eslint/parser": "^2.0.0",
```
Comment (B3zo0) [inline examples/sn-react-typescript-boilerplate/package.json]:
```suggestion
```

File: examples/sn-react-typescript-boilerplate/package.json [hunk 6]
```diff
@@ -0,0 +1,66 @@
...
+    "awesome-typescript-loader": "^5.2.1",
+    "css-loader": "^3.0.0",
+    "enzyme": "^3.9.0",
```
Comment (B3zo0) [inline examples/sn-react-typescript-boilerplate/package.json]:
```suggestion
```

File: examples/sn-react-typescript-boilerplate/package.json [hunk 7]
```diff
@@ -0,0 +1,66 @@
...
+    "eslint": "^6.0.1",
```
Comment (B3zo0) [inline examples/sn-react-typescript-boilerplate/package.json]:
```suggestion
```

File: examples/sn-react-typescript-boilerplate/package.json [hunk 8]
```diff
@@ -0,0 +1,66 @@
...
+    "file-loader": "^4.0.0",
+    "html-webpack-plugin": "^3.2.0",
+    "jest": "^24.8.0",
+    "prettier": "^1.17.1",
```
Comment (B3zo0) [inline examples/sn-react-typescript-boilerplate/package.json]:
```suggestion
```

File: examples/sn-react-typescript-boilerplate/package.json [hunk 9]
```diff
@@ -0,0 +1,66 @@
...
+    "@types/react-dom": "^16.8.4",
+    "@typescript-eslint/eslint-plugin": "^2.0.0",
+    "@typescript-eslint/parser": "^2.0.0",
+    "awesome-typescript-loader": "^5.2.1",
+    "css-loader": "^3.0.0",
+    "enzyme": "^3.9.0",
+    "eslint": "^6.0.1",
+    "file-loader": "^4.0.0",
+    "html-webpack-plugin": "^3.2.0",
+    "jest": "^24.8.0",
```
Comment (B3zo0) [inline examples/sn-react-typescript-boilerplate/package.json]:
```suggestion
```

File: examples/sn-react-typescript-boilerplate/package.json [hunk 10]
```diff
@@ -0,0 +1,66 @@
...
+    "prettier": "^1.17.1",
+    "rimraf": "^3.0.0",
+    "source-map-loader": "^0.2.4",
+    "style-loader": "^1.0.0",
+    "ts-config-webpack-plugin": "^1.3.1",
+    "ts-jest": "^24.0.2",
```
Comment (B3zo0) [inline examples/sn-react-typescript-boilerplate/package.json]:
```suggestion
```

File: examples/sn-react-typescript-boilerplate/typings/index.d.ts [hunk 11]
```diff
@@ -0,0 +1,5 @@
+declare module '*.md'
```
Comment (B3zo0) [inline examples/sn-react-typescript-boilerplate/typings/index.d.ts]: Please remove this folder

File: examples/sn-react-typescript-boilerplate/tsconfig.json [hunk 12]
```diff
@@ -0,0 +1,17 @@
...
+    "composite": false,
+    "jsx": "react",
+    "rootDir": "./src",
+    "forceConsistentCasingInFileNames": true,
+    "experimentalDecorators": true,
+    "emitDecoratorMetadata": true
+  },
+  "include": [
+    "src",
+    "./typings"
```
Comment (B3zo0) [inline examples/sn-react-typescript-boilerplate/tsconfig.json]:
```suggestion
    "../../typings"
```

**Comment by sumirart** [comment]
Thanks, but sorry I can't add suggestion to batch because I already solve all the conversation.

**Comment by sumirart** [comment]
I already submit the form, thanks!

**Pull request merged**